test/unit/StakingPool/calculatePremium.js

```diff
@@ -0,0 +1,480 @@
+    useFixedPrice: false,
+  };
+
+  const productInitializationParams = [
+    {
+      productId: 0,
+      weight: 100, // 1.00
+      initialPrice: coverProductTemplate.initialPriceRatio,
+      targetPrice: 200, // 2%}
+    },
+  ];
+
+  beforeEach(async function () {
+    const { stakingPool, cover } = this;
+    const { defaultSender: manager } = this.accounts;
+    const [staker] = this.accounts.members;
+    const productId = 0;
+    // Set global product and product type
+    await cover.setProduct(coverProductTemplate, productId);
+    await cover.setProductType({ claimMethod: 1, gracePeriod: daysToSeconds(7) }, productId);
+
+    // Initialize staking pool
+    const poolId = 0;
+    const isPrivatePool = false;
+    const ipfsDescriptionHash = 'Staking pool 1';
+    const maxPoolFee = 10; // 10%
+    const initialPoolFee = 7; // 7%
+
+    await cover.initializeStaking(
+      stakingPool.address,
+      manager.address,
+      isPrivatePool,
+      initialPoolFee,
+      maxPoolFee,
+      productInitializationParams,
+      poolId,
+      ipfsDescriptionHash,
+    );
+
+    // Deposit into pool
+    const amount = stakedNxmAmount;
+    await stakingPool.connect(staker).depositTo([
+      {
+        tokenId: 0,
+        amount,
+        destination: staker.address,
+        trancheId: (await getCurrentTrancheId()) + 4,
+      },
+    ]);
+  });
+
+  async function moveDaysForward(days) {
+    let { timestamp } = await ethers.provider.getBlock('latest');
+    timestamp += daysToSeconds(days);
+    await setNextBlockTime(timestamp);
+    await mineNextBlock();
+  }
+
+  async function setNextBlockDaysForward(days) {
+    let { timestamp } = await ethers.provider.getBlock('latest');
+    timestamp += daysToSeconds(days);
+    await setNextBlockTime(timestamp);
+  }
+
+  it('should correctly calculate the premium and price for year long cover', async function () {
+    const { stakingPool, cover } = this;
+    const { GLOBAL_CAPACITY_RATIO, NXM_PER_ALLOCATION_UNIT, INITIAL_PRICE_DENOMINATOR } = this.config;
+
+    const product = await stakingPool.products(productId);
+    const initialPrice = BigNumber.from(coverProductTemplate.initialPriceRatio);
+    expect(product.nextPrice).to.be.equal(initialPrice);
+
+    const { totalCapacity } = await stakingPool.getActiveTrancheCapacities(
+      buyCoverParamsTemplate.productId,
+      GLOBAL_CAPACITY_RATIO,
+      coverProductTemplate.capacityReductionRatio,
+    );
+    const expectedPremium = buyCoverParamsTemplate.amount.mul(initialPrice).div(INITIAL_PRICE_DENOMINATOR);
+    const priceBump = await calculatePriceBump(
+      buyCoverParamsTemplate.amount.div(NXM_PER_ALLOCATION_UNIT),
+      this.config.PRICE_BUMP_RATIO,
+      totalCapacity,
+    );
+
+    {
+      // buy cover and check premium + new price
+      const buyCoverParams = { ...buyCoverParamsTemplate, period: daysToSeconds(365) };
+      await cover.allocateCapacity(buyCoverParams, coverId, stakingPool.address);
```

Comment:
the test here is

`calculatePremium`

we have a function now called just that which is pure.

The tests in this file shoul concern just that function.

allocateCapacity should be tested separately.